stdlib/public/runtime/ObjCRuntimeGetImageNameFromClass.mm

```diff
@@ -251,6 +324,9 @@ static void patchGetImageNameInImage(const struct mach_header *mh,
   const void *newImplementationAddr =
       reinterpret_cast<const void *>(&patchedGetImageNameFromClassForOldOSs);
   patchLazyPointers(mh, "_class_getImageName", newImplementationAddr);
+#if PATCH_NSBUNDLE
+  patchNSBundle();
```

Comment:
Oh, bah. I realized that since libswiftCore links against Foundation anyway it'd be okay to do this eagerly once rather than running it in the "new image loaded" callback.

Reply:
I guess we're trying not to rely on linking to Foundation here, so okay.